test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore/NavigationPropertyOnComplexType/NavigationPropertyOnComplexTypeTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.Test.E2E.AspNet.OData.Common.Execution;
+using Xunit;
+
+namespace Microsoft.Test.E2E.AspNet.OData.NavigationPropertyOnComplexType
+{
+    public class NavigationPropertyOnComplexType : WebHostTestBase
+    {
+        private const string PeopleBaseUrl = "{0}/odata/people";
+
+        public NavigationPropertyOnComplexType(WebHostTestFixture fixture)
+            : base(fixture)
+        {
+        }
+
+        protected override void UpdateConfiguration(WebRouteConfiguration configuration)
+        {
+            configuration.AddControllers(typeof(PeopleController));
+            configuration.JsonReferenceLoopHandling =
+                Newtonsoft.Json.ReferenceLoopHandling.Ignore;
+            configuration.MaxTop(2).Expand().Select().OrderBy().Filter();
+            configuration.MapODataServiceRoute("odata", "odata", ModelGenerator.GetConventionalEdmModel());
+        }
+
+        [Theory]
+        [InlineData(PeopleBaseUrl + "(1)/Location/ZipCode", "{\"Zip\":98052,\"City\":\"Redmond\",\"State\":\"Washington\"}")]
```

Comment:
>Location/ZipCode [](start = 41, length = 16)

Make sure "People(1)?$expand=Location/ZipCode "  work

Reply:
Currently, it does not work. Having a path in expand throws a validation error. I believe it is a slightly different feature to enable that. Once path is supported, I don't think there should be other problems.

---
In reply to: [250465447](https://github.com/OData/WebApi/pull/1738#discussion_r250465447) [](ancestors = 250465447)